discover/gpu.go

```diff
@@ -92,7 +97,7 @@ func initCudaHandles() *cudaHandles {
 	}
 
 	slog.Debug("searching for GPU discovery libraries for NVIDIA")
-	var cudartMgmtPatterns []string
+	cudartMgmtPatterns := []string{}
```

Comment:
I think former version of this is cleaner unless you purposely want an empty slice vs `nil`